v2/internal/frontend/assetserver/assethandler.go

```diff
@@ -33,8 +32,7 @@ type assetHandler struct {
 	retryMissingFiles bool
 }
 
-func NewAssetHandler(ctx context.Context, options *options.App) (http.Handler, error) {
-	vfs := options.Assets
+func NewAssetHandler(ctx context.Context, vfs iofs.FS, AssetsHandler http.Handler) (http.Handler, error) {
```

Comment:
The assetsHandler parameter should shard with a lower case.
```suggestion
func NewAssetHandler(ctx context.Context, vfs iofs.FS, assetsHandler http.Handler) (http.Handler, error) {
```